### PR TITLE
Support for github Authentication Tokens

### PR DESCRIPTION
**Pull request by orta**
Helps the discussion in #14

Sorry this took a few weeks, but I needed fresh perspective - once I read https://github.com/mattbrictson/chandler/issues/14#issuecomment-245418267 I really started to grok your testing systems. Thanks for that! Not used mini test 👍 - it's _fast_.

### CODE REVIEW

**Comment by orta** [comment]
The irony of not forgetting a CHANGELOG entry is definitely not lost on me. [Danger](https://github.com/danger/danger/) can help there BTW ;)

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ chandler is in a pre-1.0 state. This means that its APIs and behavior are subjec
 
 ## [Unreleased][]
 
-* Your contribution here!
```
Comment (mattbrictson) [inline CHANGELOG.md]: Could you re-add the "Your contribution here!" line? I like to keep it there as a signpost for future contributors.

File: lib/chandler/configuration.rb [hunk 2]
```diff
@@ -23,9 +23,28 @@ def git
       @git ||= Chandler::Git.new(:path => git_path, :tag_mapper => tag_mapper)
     end
 
+    def octokit
+      @octokit ||= Octokit::Client.new(client_options)
+    end
+
+    def client_options
```
Comment (mattbrictson) [inline lib/chandler/configuration.rb]: Nitpick: what about renaming this to `octokit_options`?

**Comment by mattbrictson** [comment]
> The irony of forgetting a CHANGELOG entry is definitely not lost on me. Danger can help there BTW ;)

Good point! In fact I've set it up for one of my other projects already ([airbrussh](https://github.com/mattbrictson/airbrussh)) and would like to do it here as well, but I'm not sure I want to set up another bot. Can I use the @dangerci user as my bot (looks like I can steal the API key from your Travis logs 😉) or do encourage each repo that uses Danger set up their own bot account?

**Comment by orta** [comment]
I generally discourage re-using that, but I know a few are using it, so long as the account doesn't get banned - I'm totally fine with it 👍 

Will edit PR now, might make that same CHANGELOG change on my own projects

**Comment by mattbrictson** [comment]
Awesome, thanks again for the PR! 🙇

**Pull request merged**